presto-router/src/test/java/com/facebook/presto/router/TestSelectors.java

```diff
@@ -0,0 +1,178 @@
+import com.google.inject.Module;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+import javax.servlet.http.HttpServletRequest;
+
+import java.io.File;
+import java.io.FileOutputStream;
+import java.net.URI;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.List;
+import java.util.Optional;
+
+import static com.facebook.presto.client.PrestoHeaders.PRESTO_CLIENT_TAGS;
+import static com.facebook.presto.client.PrestoHeaders.PRESTO_SOURCE;
+import static com.facebook.presto.client.PrestoHeaders.PRESTO_USER;
+import static java.nio.charset.StandardCharsets.UTF_8;
+import static org.testng.Assert.assertEquals;
+
+public class TestSelectors
+{
+    private List<TestingPrestoServer> prestoServers;
+    private ClusterManager clusterManager;
+    private LifeCycleManager lifeCycleManager;
+
+    @BeforeClass
+    public void setup()
+            throws Exception
+    {
+        String configTemplate = new String(
+                Files.readAllBytes(Paths.get(this.getClass().getClassLoader().getResource("selector-test-router-template.json").getPath())));
+
+        File configFile = File.createTempFile("router", "json");
+
+        Logging.initialize();
+
+        // set up server
+        ImmutableList.Builder<TestingPrestoServer> builder = ImmutableList.builder();
+        for (int i = 0; i < 3; ++i) {
+            TestingPrestoServer server = new TestingPrestoServer();
+            server.installPlugin(new TpchPlugin());
+            server.createCatalog("tpch", "tpch");
+            server.refreshNodes();
+            builder.add(server);
+        }
+
+        prestoServers = builder.build();
+
+        for (TestingPrestoServer s : prestoServers) {
+            configTemplate = configTemplate.replaceFirst("\\$\\{SERVERS}", String.format("[ \"%s\" ]", s.getBaseUrl().toString()));
+        }
+
+        FileOutputStream fileOutputStream = new FileOutputStream(configFile);
+        fileOutputStream.write(configTemplate.getBytes(UTF_8));
+
+        fileOutputStream.close();
+
+        Bootstrap app = new Bootstrap(ImmutableList.<Module>builder()
+                .add(new TestingNodeModule())
+                .add(new TestingHttpServerModule())
+                .add(new JsonModule())
+                .add(new JaxrsModule(true))
+                .add(new ServerSecurityModule())
+                .add(new RouterModule())
+                .build());
+
+        Injector injector = app.doNotInitializeLogging().setRequiredConfigurationProperty("router.config-file", configFile.getAbsolutePath()).quiet().initialize();
+        clusterManager = injector.getInstance(ClusterManager.class);
+        lifeCycleManager = injector.getInstance(LifeCycleManager.class);
+    }
+
+    @AfterClass(alwaysRun = true)
+    public void tearDownServer()
+            throws Exception
+    {
+        for (TestingPrestoServer prestoServer : prestoServers) {
```

Comment:
we should check if `prestoServers` is null before this. Same with `lifecycleManager`. If they are null, make the actions a no-op